p2p/p2p.go

```diff
@@ -74,17 +78,30 @@ func New(addr, userAgent, peers, privKeyStr string, feederNode bool, bc *blockch
 	// Todo: try to understand what will happen if user passes a multiaddr with p2p public and a private key which doesn't match.
 	// For example, a user passes the following multiaddr: --p2p-addr=/ip4/0.0.0.0/tcp/7778/p2p/(SomePublicKey) and also passes a
 	// --p2p-private-key="SomePrivateKey". However, the private public key pair don't match, in this case what will happen?
-	return NewWithHost(p2pHost, peers, feederNode, bc, snNetwork, log)
+	return NewWithHost(p2pHost, peers, feederNode, bc, snNetwork, log, database)
 }
 
 func NewWithHost(p2phost host.Host, peers string, feederNode bool, bc *blockchain.Blockchain, snNetwork *utils.Network,
-	log utils.SimpleLogger,
+	log utils.SimpleLogger, database db.DB,
 ) (*Service, error) {
-	peersAddrInfoS := []peer.AddrInfo{}
+	var (
+		peersAddrInfoS []peer.AddrInfo
+		storePeers     []peer.AddrInfo
+		err            error
+	)
+
+	storePeers, err = loadPeers(database)
+	if err != nil {
+		log.Warnw("Failed to load peers", "err", err)
+	}
+
+	peersAddrInfoS = append(peersAddrInfoS, storePeers...)
```

Comment:
we can further simplify it to :
```
peersAddrInfoS, err := loadPeers(database)
if err != nil {
  ...
}
```
in case of an error `peersAddrInfoS` will be nil and that's fine (there will be no segfaults) for `append(...)` and `someFunc(peersAddrInfoS...)`